setup.py

```diff
@@ -6,7 +6,8 @@
     'six>=1.10.0',
     'graphql-core>=2,<3',
     'promise>=2.0,<3',
-    'requests>=2.12,<3'
+    'requests>=2.12,<3',
+    'websockets>=8.1,<9'
```

Comment:
From pypi, websockets support Python 3.6.1 and higher. This one should be appended to the `install_requires` list. See line 20 for an example with `pytest-asyncio`.

Reply:
Done. Now tests are passing. Will need to add tests to raise the coverage now.

Reply:
Also now that I understand asyncio a little better I think it is best to use Futures intead of Queues for the listeners. Will need to change that also so don't accept this PR just yet.

Reply:
No problem mate :)

Reply:
In fact queues are really needed for slow consumers ...